appserver/web/weld-integration/src/main/java/org/glassfish/weld/ACLSingletonProvider.java

```diff
@@ -108,10 +103,27 @@ public T get(String id) {
             if (instance == null) {
                 throw new IllegalStateException("Singleton not set for " + appClassLoader);
             }
-
+            LOG.log(DEBUG, () -> "get(" + id + ") - found " + instance);
             return instance;
         }
 
+        @Override
+        public boolean isSet(String id) {
+            return store.containsKey(getClassLoader());
+        }
+
+        @Override
+        public void set(String id, T object) {
+            LOG.log(DEBUG, () -> "set(id=" + id + ", object=" + object + ")");
+            store.put(getClassLoader(), object);
+        }
+
+        @Override
+        public void clear(String id) {
+            LOG.log(DEBUG, () -> "clear(id=" + id + ")");
+            store.remove(getClassLoader());
+        }
+
```

Comment:
A bit strange that `id`s are completely ignored in those methods. Could the parameters be renamed to express this intention?

Reply:
Nope, the Singleton class is Weld's interface. The funny thing is that the id is somehow useful, however I doubt it is guaranteed that it is unique. Probably that is the reason why somebody uses the classloader:
`set(id=DeploymentDescriptorTests_web, object=org.jboss.weld.Container@3bf27b0)]]`

Reply:
>> Could the parameters be renamed to express this intention?

> Nope, the Singleton class is Weld's interface. 

But Weld does not have to know what is done to the parameter name in this implementation.
I find it also a little disturbing that it's logged like `id` values matter, while they don't.

Reply:
So what is your idea? I don't like it too, it smells, I see in logs it is probably always the application name and maybe it could be used as a key instead of classloader, but ... I can 

* delete all these logs (but then I cannot read anything useful in logs)
* change `id=` to `application=` (but that can change one day without warning, it is rather guessing what it really is)
* switch from CL to the id (but that would be quite risky experiment now)
* keep it with grinding teeth (confusing as you wrote)
* add `", but using thread classloader as an id instead"`

I did this because it was the simplest way to get at least something.

Reply:
Do not consider my comment to be blocking :) (But I've not reviewed everything yet.).

I was only thinking about change `id` -> `ignoredId` - and keep everything else as-is. Just for me, reading this somewhere in the future.

I'm for marking this one detail as `resolved`, and leave it on _to think about list_.

Reply:
Finally I added at least the comment to the log.